archive_upload/handlers/dsmc_handlers.py

```diff
@@ -0,0 +1,754 @@
+            """
+            name = os.path.basename(tarinfo.name)
+            # The name field contains the path to the file relative to the
+            # root dir of the archive, i.e. the path starts with "./".
+            # Therefore the second element in the list split on "/"
+            # will be the first subdir (if any) inside the archive.
+            first_dir = tarinfo.name.split("/")[1]
+
+            # Don't include the file if it matches our list of
+            # files to exclude, or if the first dir in its path
+            # matches one of the dir names in our exception list.
+            for exclude in exclude_from_tarball:
+                if exclude == name or exclude == first_dir:
+                    return None
+
+            return tarinfo
+
+        exclude_from_tarball = self.config["exclude_from_tarball"]
+        log.info("Creating tarball {}...".format(tarball_path))
+
+        with tarfile.open(name=tarball_path, mode="w:gz", dereference=True) as tar:
+            tar.add(path_to_archive, arcname="./", recursive=True, filter=exclude_content)
+
+        log.info("Removing files from {} that were added to {}".format(path_to_archive_root, tarball_path))
+
+        # Remove files that we added to the tarball.
+        with tarfile.open(tarball_path) as tar:
+            for member in tar.getmembers():
+                try:
+                    filepath = os.path.normpath(os.path.join(path_to_archive_root, archive, member.name))
+
+                    if os.path.isfile(filepath):
+                        os.remove(filepath)
+                    elif member.name != "." and os.path.isdir(filepath):
+                        self._rm_empty_dirs(filepath)
+
+                except OSError, e:
+                    msg = "Error when creating archive tarball. Could not remove file {}: {}".format(filepath, e)
+                    self.abort(msg)
+                    return
+
+        response_data = {"service_version": version, "state": State.DONE}
+        self.set_status(200, reason="Finished creating the tarball")
+        self.write_object(response_data)
+
+
+class StatusHandler(BaseDsmcHandler):
+    """
+    Get the status of one or all jobs.
+    """
+
+    def get(self, job_id):
+        """
+        Get the status of the specified job_id, or if now id is given, the
+        status of all jobs.
+        :param job_id: to check status for (set to empty to get status for all)
+        """
+
+        if job_id:
+            status = {"state": self.runner_service.status(job_id)}
+        else:
+            # TODO: Update the correct status for all jobs; the filtering in jobrunner doesn't work here.
+            all_status = self.runner_service.status_all()
+            status_dict = {}
+            for k, v in all_status.iteritems():
+                status_dict[k] = {"state": v}
+            status = status_dict
+
+        self.write_json(status)
+
+class StopHandler(BaseDsmcHandler):
```

Comment:
Perhaps the `StopHandler` should be removed for now as it is not used?

Reply:
Guess we can comment it out at least.